[18.0][MIG] barcodes_generator_product: Migration to 18.0

Kimkhoi3010 opened the pull request:
Depend on:

- https://github.com/Kimkhoi3010/stock-logistics-barcode/pull/1

legalsylvain committed:
create new module 'barcodes_generator_abstract

legalsylvain committed:
New module barcode_generator_product

oca-transbot committed:
OCA Transbot updated translations from Transifex

lasley committed:
[MIG] barcodes_generator_abstract: Upgrade to v10
* Rename manifest
* Change openerp references to odoo
* Bump version
* Alphabetize imports
* pyBarcode is no longer maintained, switch to viivakoodi fork (https://bitbucket.org/whitie/python-barcode/issues/16/pypi-08-release-request#comment-33978213)

lasley committed:
[MIG] barcodes_product_generator: Upgrade to v10
* Rename manifest
* Change openerp references to odoo
* Bump version
* Fix spelling errors and add product avatar to icon
* Remove product_barcode_generator

lasley committed:
[10.0][IMP] barcodes_generator_abstract: Add automated generator (#77)

Add an option to allow for automatic generation of barcode when a record of mixin model is created

lasley committed:
[10.0][FIX] barcodes_generator_abstract: Caching issue (#103)

* [FIX] barcodes_generator_abstract: Caching issue
* Make cache method return ids & move the recordset method out of cache to fix #93

phatnguyenuit committed:
[MIG] barcodes_generator_abstract: Migration to 11.0

phatnguyenuit committed:
[MIG] barcodes_generator_product: Migration to 11.0

legalsylvain committed:
[FIX] do not copy barcode_base during duplication

misern2 committed:
11.0 [MIG] barcode_generator_lot

cvinh committed:
[MIG] barcodes_generator_abstract: Migration to 12.0

cvinh committed:
[MIG] barcodes_generator_product: Migration to 12.0

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

AndreVolksdorf committed:
Translated using Weblate (German)

Currently translated at 29.6% (8 of 27 strings)

Translation: stock-logistics-barcode-12.0/stock-logistics-barcode-12.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-12-0/stock-logistics-barcode-12-0-barcodes_generator_abstract/de/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 14.8% (4 of 27 strings)

Translation: stock-logistics-barcode-12.0/stock-logistics-barcode-12.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-12-0/stock-logistics-barcode-12-0-barcodes_generator_abstract/sl/

CarlosRoca13 committed:
[ADD] stock_picking_product_barcode_report: New module that allows you to select how many labels to print

francesco-ooops committed:
Added translation using Weblate (Italian)

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 46.4% (13 of 28 strings)

Translation: stock-logistics-barcode-12.0/stock-logistics-barcode-12.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-12-0/stock-logistics-barcode-12-0-barcodes_generator_abstract/it/

alexis-via committed:
barcodes_generator_abstract: Fix warning

Improve form view of barcode rules
Improve README
Auto-add admin to the group

CarlosRoca13 committed:
[FIX] stock_picking_product_barcode_report: Avoid empty data on wizard

CarlosRoca13 committed:
[IMP] barcodes_generator_abstract: black, isort, prettier

CarlosRoca13 committed:
[MIG] barcodes_generator_abstract: Migration to v13.0

CarlosRoca13 committed:
[IMP] barcodes_generator_product: black, isort, prettier

oca-travis committed:
[UPD] Update barcodes_generator_abstract.pot

CarlosRoca13 committed:
[MIG] barcodes_generator_product: Migration to v13.0

CarlosRoca13 committed:
[IMP] stock_picking_product_barcode_report: Add the possibility to define other report format

sergio-teruel committed:
[IMP] stock_picking_product_barcode_report: Better inheritance of method to select move lines to print

Zar21 committed:
[IMP] barcodes_generator_abstract: black, isort, prettier

Zar21 committed:
[MIG] barcodes_generator_abstract: Migration to 14.0

oca-travis committed:
[UPD] Update barcodes_generator_abstract.pot

OCA-git-bot committed:
[UPD] README.rst

hugho-ad committed:
[IMP] barcodes_generator_product: black, isort, prettier

hugho-ad committed:
[MIG] barcodes_generator_product: Migration to 14.0

hugho-ad committed:
[FIX] barcodes_generator_abstract: Duplicate implicit target name: note

OCA-git-bot committed:
[UPD] README.rst

Commit:
[FIX] stock_picking_product_barcode_report: import order

When updating the addon after https://github.com/OCA/stock-logistics-barcode/pull/328, if the database had any transient record of the `stock.picking.print` model, the update failed with:

    psycopg2.errors.UndefinedColumn: column res_company.barcode_default_report does not exist

To fix that problem, import wizard after models.

@Tecnativa TT28969

kevinkhao committed:
[FIX] remove redundant README fragment, fix dependencies in manifest

OCA-git-bot committed:
barcodes_generator_abstract 14.0.1.0.1

OCA-git-bot committed:
[UPD] README.rst

sergio-teruel committed:
[IMP] stock_picking_product_barcode_report: Allow to print quant package as barcode labels

sergio-teruel committed:
[IMP] stock_picking_product_barcode_report: do not check product barcode for move lines when user print the package labels

sergio-teruel committed:
[IMP] stock_picking_product_barcode_report: do not check product barcode for move lines when user print the package labels

sergio-teruel committed:
[IMP] stock_picking_product_barcode_report: Increase font size of package name

sergio-teruel committed:
[IMP] stock_picking_product_barcode_report: Translate

CarlosRoca13 committed:
[IMP] stock_picking_product_barcode_report: Increase font size of product info

sebastienbeau committed:
[FIX] fix UI, hide barcode field on template when having variants

sebastienbeau committed:
[FIX] add missing view on product easy form view

sergio-teruel committed:
[IMP] stock_picking_product_barcode_report: hide hour in life date

sergio-teruel committed:
[IMP] stock_picking_product_barcode_report: Print quant package content labels
TT34791

pedrobaeza committed:
[IMP+FIX] Update dotfiles + switch to GH actions + manifest fix

SicurSam committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (19 of 19 strings)

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-barcodes_generator_product
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-barcodes_generator_product/it/

SicurSam committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (28 of 28 strings)

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-barcodes_generator_abstract/it/

AshishHirapara committed:
[IMP][14.0] barcodes_generator_abstract

updated images path on configuration readme file

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
barcodes_generator_abstract 14.0.1.0.2

legalsylvain committed:
[MIG] barcodes_generator_abstract from 14 to V16

legalsylvain committed:
[MIG] barcodes_generator_product from 14.0 to 16.0

Commit:
[UPD] Update barcodes_generator_abstract.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/

CarlosRoca13 committed:
[IMP] stock_picking_product_barcode_report: black, isort, prettier

CarlosRoca13 committed:
[MIG] stock_picking_product_barcode_report: Migration to 15.0

Commit:
[UPD] Update barcodes_generator_abstract.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (20 of 20 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/it/

FranciscoFactorLibre committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (20 of 20 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/es/

sergio-teruel committed:
[IMP] stock_picking_product_barcode_report: Manage packages and better filter origin moves

ilyasProgrammer committed:
[FIX] barcodes_generator_product: #516

ilyasProgrammer committed:
[FIX] barcodes_generator_product: #516

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
barcodes_generator_abstract 16.0.1.0.2

CarlosRoca13 committed:
[IMP] stock_picking_product_barcode_report: Posibility to print labels from HTML field

legalsylvain committed:
[IMP] barcodes_generator_product : Prevent user to create barcode.rule via product forms, to avoid to have barcode.rule not related to barcode.nomenclature. (Such rule will not be accessible after, because there is no menu entry to access to barcode.rule

legalsylvain committed:
[IMP] barcode_generator_abstract : UX : set the field barcode_rule.generate_model required if generate_type is defined

legalsylvain committed:
[IMP] barcodes_generator_abstract : UX : Do not display barcode_rule.generate_automate for non 'sequence' generation

legalsylvain committed:
[IMP] barcodes_generator_abstract : UX : Adds a message to specify that sequence creation can be automatic.

legalsylvain committed:
[IMP] barcodes_generator_abstract : UX : display barcode.rule new field on barcode.nomenclature form view

legalsylvain committed:
[REF] barcodes_generator_abstract : Refactor. Do not mix automatic generation and default value concepts:
- allow to have multiple automatic barcode.rule for a same model
- Simplify a lot the code base
- remove useless cache clearing

legalsylvain committed:
[REF] barcode_generate_abstract : Add more tests

legalsylvain committed:
[REF] Improve fields description

legalsylvain committed:
[REF] Improve fields description

legalsylvain committed:
[REF] barcodes_generator_abstract : Use external_dependencies_override in setup file to correctly define python library

legalsylvain committed:
[REF] barcodes_generator_product : use new way to handle related fields between product.product and product.template. (using _get_related_fields_variant_template, set compute / inverse function on fields, etc...)

Commit:
[UPD] Update barcodes_generator_abstract.pot

OCA-git-bot committed:
barcodes_generator_abstract 16.0.3.0.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/

legalsylvain committed:
[FIX] barcodes_generator_abstract : align create algorithm with write algorightm

OCA-git-bot committed:
barcodes_generator_abstract 16.0.3.0.1

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (22 of 22 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/es/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (17 of 17 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_product
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_product/es/

OCA-git-bot committed:
[UPD] README.rst

mymage committed:
Translated using Weblate (Italian)

Currently translated at 70.5% (12 of 17 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_product
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_product/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 77.2% (17 of 22 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/it/

RodrigoBM committed:
[FIX] barcodes_generator_abstract: get the real integer code for all coding types
if you use for example the gs1-128 coding, this coding cannot be saved in odoo directly.
>>> import barcode
>>> EAN = barcode.get_barcode_class('gs1_128')
>>> str(EAN('0000'))
'ñ0000'
We have to call the function get_fullcode which is a code that we can save in Odoo.
>>> import barcode
>>> EAN = barcode.get_barcode_class('gs1_128')
>>> str(EAN('0000').get_fullcode())
'0000'

OCA-git-bot committed:
[BOT] post-merge updates

pedrobaeza committed:
[FIX] barcodes_generator_*: Linter fixes

pedrobaeza committed:
[FIX] barcodes_generator_*: Linter fixes

OCA-git-bot committed:
[BOT] post-merge updates

LESTRAT21 committed:
Translated using Weblate (French)

Currently translated at 94.1% (16 of 17 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_product
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_product/fr/

CarlosRoca13 committed:
[IMP] stock_picking_product_barcode_report: Remove space under barcode

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (22 of 22 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (17 of 17 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_product
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_product/it/

sergio-teruel committed:
[FIX] stock_picking_product_barcode_report: Write on stock move lines linked due to related uom field

CarlosRoca13 committed:
[IMP] stock_picking_product_barcode_report: Use better the space defined by the image

CarlosRoca13 committed:
[IMP] stock_picking_product_barcode_report: Add possibility to print labels from stock.move.line

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (22 of 22 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/it/

HekkiMelody committed:
[IMP] barcodes_generator_product: clarify documentation

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (22 of 22 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-barcodes_generator_abstract/it/

sergio-teruel committed:
[IMP] stock_picking_product_barcode_report: Allow print labels depends on units included into a product packaging
TT47545

CarlosRoca13 committed:
[IMP] stock_picking_product_barcode_report: Allow force qty using context

CarlosRoca13 committed:
[IMP] stock_picking_product_barcode_report: Allow print from stock_quants

legalsylvain committed:
[FIX] barcodes_generator_product : set widget selection on barcode_rule_id, to avoid an OWL error 'TypeError: record is undefined'.
Remark : It doesn't makes sense to edit barcode rule via product / template, as it should be managed via barcode nomenclature form view

sergio-teruel committed:
[IMP] stock_picking_product_barcode_report: Add product packaging to wizard. New base report label to be extended by other modules. Label qty computed

sergio-teruel committed:
[IMP] stock_picking_product_barcode_report: Allow to select language on wizard to print labels.
TT48582

isaako34 committed:
[IMP] barcodes_generator_abstract: set 'python-barcode' as external_dependencies

Since using the PyPI package has been the recommended way since v12.0
(see https://github.com/odoo/odoo/pull/25549), this commit sets
'python-barcode' as an external dependency and removes the
'odoo_addon' key from setup.py

OCA-git-bot committed:
[BOT] post-merge updates

isaako34 committed:
[IMP] : pre-commit auto fixes

isaako34 committed:
[MIG] barcodes_generator_abstract: Migration to 17.0

- Migrate to 17.0
- Add the package 'odoo_test_helper' to test-requirements.txt

Commit:
[UPD] Update barcodes_generator_abstract.pot

OCA-git-bot committed:
[BOT] post-merge updates

isaako34 committed:
[IMP] barcodes_generator_product: pre-commit auto fixes

isaako34 committed:
[MIG] barcodes_generator_product: Migration to 17.0

- Migrate views using the module `views_migration_v17`
- Remove the 'priority' field in views.
- Remove the 'replace' attribute from views and apply the 'after'
  attribute instead.
- Migrate setUp to setUpClass.
- Apply command syntax.
- Fix the unordered list in DESCRIPTION.md and README.rst files.

Commit:
[UPD] Update barcodes_generator_product.pot

OCA-git-bot committed:
[BOT] post-merge updates

sergio-teruel committed:
[IMP] stock_picking_product_barcode_report: Add helper method to create labels from other models

CarlosRoca13 committed:
[MIG] stock_picking_product_barcode_report: Migration to 16.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-stock_picking_product_barcode_report
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-stock_picking_product_barcode_report/

mymage committed:
Added translation using Weblate (Italian)

kevinkhao committed:
[17.0] stock_picking_product_barcode_report: Migration to 17.0

GabbasovDinar committed:
[IMP] product_multi_barcode: Performance

Improve search speed when searching products by name.

Task: 4183

Commit:
[IMP] product_multi_barcode: readme improved

Improve Readme USAGE, DESCRIPTIOM files and add the screenshot to be more user-friendly

etobella committed:
[FIX] stock_picking_product_barcode_report: Use barcode_type, as type was deprecated

Kimkhoi3010 committed:
[IMP] barcodes_generator_abstract: pre-commit auto fixes

Kimkhoi3010 committed:
[MIG] barcodes_generator_abstract: Migration to 18.0

Kimkhoi3010 committed:
[IMP] barcodes_generator_product: pre-commit auto fixes

Kimkhoi3010 committed:
[MIG] barcodes_generator_product: Migration to 18.0

nguyenminhchien commented on barcodes_generator_product/tests/test_barcodes_generator_product.py:
```diff
@@ -0,0 +1,78 @@
+from odoo.tests.common import TransactionCase
+
+
+class Tests(TransactionCase):
```
```suggestion
class Tests(BaseCommon):
```


Use BaseCommon as base test class to reduce overhead from tracking and boost the test suite.

nguyenminhchien replied:
reference: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-18.0#tasks-to-do-in-the-migration

Anxo82 committed:
[IMP] stock_picking_product_barcode_report: pre-commit auto fixes

bo-eska committed:
Translated using Weblate (Turkish)

Currently translated at 94.1% (16 of 17 strings)

Translation: stock-logistics-barcode-18.0/stock-logistics-barcode-18.0-product_multi_barcode
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-18-0/stock-logistics-barcode-18-0-product_multi_barcode/tr/

Anxo82 committed:
[FIX] stock_picking_product_barcode_report: Fix parameter name in barcode generation (type → barcode_type)

celm1990 committed:
[IMP] stock_picking_product_barcode_report: pre-commit auto fixes

carlosdauden committed:
[FIX] stock_picking_product_barcode_report: Error printing from quants because doc model is interpreted as stock.picking.line.print

carlos-lopez-tecnativa committed:
[MIG] stock_picking_product_barcode_report: Migration to version 18.0

OCA-git-bot committed:
Merge PR #705 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update stock_picking_product_barcode_report.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #681 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update barcodes_generator_abstract.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #719 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

bo-eska committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (17 of 17 strings)

Translation: stock-logistics-barcode-18.0/stock-logistics-barcode-18.0-product_multi_barcode
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-18-0/stock-logistics-barcode-18-0-product_multi_barcode/tr/

CarlosRoca13 committed:
[UPD] Update copier template

pedrobaeza committed:
Merge pull request #726 from Tecnativa/18.0-UPD-copier_template

[18.0][UPD] Update copier template
